bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -244,12 +278,12 @@ where
 		let inbound_queue_pallet_index = InboundQueuePalletInstance::get();
 
 		let mut instructions = vec![
-			ReceiveTeleportedAsset(total_fee_asset.into()),
-			BuyExecution { fees: asset_hub_fee_asset, weight_limit: Unlimited },
 			DescendOrigin(PalletInstance(inbound_queue_pallet_index).into()),
 			UniversalOrigin(GlobalConsensus(network)),
-			ReserveAssetDeposited(asset.clone().into()),
-			ClearOrigin,
+			// Todo: Hardcode for the POC and will pass original sender from Ethereum side
+			DescendOrigin([AccountKey20 { network: None, key: [1u8; 20] }].into()),
+			ReserveAssetDeposited(vec![total_fee_asset, asset.clone()].into()),
+			BuyExecution { fees: asset_hub_fee_asset, weight_limit: Unlimited },
 		];
```

Comment:
Replace `ReceiveTeleportedAsset ` with `ReserveAssetDeposit` for xcm fees. 

The reason is for the former when failed at executing the `BuyExecution { fees: asset_hub_fee_asset, weight_limit: Unlimited }` for insufficient fee, the token(weth) is not in registry yet so can't be claimed thus lost.

Though it require xcm config on AssetHub `IsReserve` to set Ethereum as trusted reserve location for the RelayToken(i.e. `DOT`).